src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using EncInterop = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue
+{
+
+    [Export(ExportContractNames.VsTypes.ProjectNodeComExtension)]
+    [AppliesTo(ProjectCapability.ManagedEditAndContinue)]
+    [ComServiceIid(typeof(IVsENCRebuildableProjectCfg))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg2))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg4))]
+    class EditAndContinueProvider : IVsENCRebuildableProjectCfg, EncInterop.IVsENCRebuildableProjectCfg2, EncInterop.IVsENCRebuildableProjectCfg4
+    {
+        private const int E_NOTIMPL = unchecked((int)0x80004001);
+        private readonly ILanguageServiceHost _host;
+
+        [ImportingConstructor]
+        public EditAndContinueProvider(
+            ILanguageServiceHost host)
+        {
+            _host = host;
+        }
+
+        #region IVsENCRebuildableProjectCfg Implementation
+        // Managed ENC always uses IVsENCRebuildableProjectCfg2. 
+        public int ENCRebuild(object in_pProgram, out object out_ppSnapshot)
+        {
+            out_ppSnapshot = null;
+            return E_NOTIMPL;
+        }
+
+        public int BelongToProject(string in_szFileName, ENC_REASON in_ENCReason, int in_fOnContinue)
+        {
+            return E_NOTIMPL;
+        }
+
+        public int ENCComplete(int in_fENCSuccess)
+        {
+            return E_NOTIMPL;
+        }
+
+        public int CancelENC()
+        {
+            return E_NOTIMPL;
+        }
+
+        public int ENCRelink([In, MarshalAs(UnmanagedType.IUnknown)] object pENCRelinkInfo)
+        {
+            return E_NOTIMPL;
+        }
+
+        public int StartDebugging()
+        {
+            return E_NOTIMPL;
+        }
+
+        public int StopDebugging()
+        {
+            return E_NOTIMPL;
+        }
+
+        public int SetENCProjectBuildOption([In] ref Guid in_guidOption, [In, MarshalAs(UnmanagedType.LPWStr)] string in_szOptionValue)
+        {
+            return E_NOTIMPL;
+        }
+        #endregion
+        #region IVsENCRebuildableProjectCfg2 Implementation
+        public int BuildForEnc([In, MarshalAs(UnmanagedType.IUnknown)] object pUpdatePE)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.ActiveProjectContext).BuildForEnc(pUpdatePE);
```

Comment:
ActiveProjectContext can be null, we should handle that and return the HR for pending.